### ISSUE TITLE
FAISS documenting filtering is broken

### ISSUE THREAD

**Issue by dpjanes** [langchain-ai/langchain#16977]
### Checked other resources

- [X] I added a very descriptive title to this issue.
- [X] I searched the LangChain documentation with the integrated search.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I am sure that this is a bug in LangChain rather than my code.

### Example Code

The issue is in `langchain/libs/community/langchain_community/vectorstores/faiss.py` at line 333

```
            if filter is not None and filter_func(doc.metadata):
                docs.append((doc, scores[0][j]))
            else:
                docs.append((doc, scores[0][j]))
```

If there's a filter, this will always add a document whether or not the function succeeds or fails (confirmed with AI: https://chat.openai.com/share/1b68d90b-ed4d-4e7d-9aff-9195acf18f96)

It should be something like:

```
            if filter is None:
                docs.append((doc, scores[0][j]))
            elif filter_func(doc.metadata):
                docs.append((doc, scores[0][j]))
```

### Description

* using FAISS with a filter just basically ignores the filter

### System Info

```
langchain==0.1.5
langchain-community==0.0.17
langchain-core==0.1.18
langchain-mistralai==0.0.3
```

**Comment by rchen19** [comment]
I believe this is fixed in v0.1.7, line 336 [here](https://github.com/langchain-ai/langchain/blob/b5d341656397de2a2935e5ddb637db5c52bac7f0/libs/community/langchain_community/vectorstores/faiss.py#L336)

```python
if filter is not None:
    if filter_func(doc.metadata):
        docs.append((doc, scores[0][j]))
else:
    docs.append((doc, scores[0][j]))
```

**Comment by dpjanes** [comment]
Functionally it looks good. A cleaner approach might be:

```python

        ## around line 308, define an always True filter_func before the rest of the code
        filter_func = lambda metadata: True
        if filter is not None:
            if isinstance(filter, dict):
            ...
 ```
 
 Then at line 336:
 
 ```python
             if filter_func(doc.metadata):
                docs.append((doc, scores[0][j]))
 ````
